js/bybit.js

```diff
@@ -4881,23 +4882,78 @@ module.exports = class bybit extends Exchange {
         return this.parseOpenInterests (result, market, since, limit);
     }
 
+    async fetchOpenInterest (symbol, params = {}) {
+        /**
+         * @method
+         * @name bybit#fetchOpenInterest
+         * @description Retrieves the open interest of a derivative trading pair
+         * @see https://bybit-exchange.github.io/docs/derivativesV3/contract/#t-dv_marketopeninterest
+         * @param {string} symbol Unified CCXT market symbol
+         * @param {object} params exchange specific parameters
+         * @param {string|undefined} params.interval "5m", 15m, 30m, 1h, 4h, 1d
+         * @param {string|undefined} params.category "linear" or "inverse"
+         * @returns {object} an open interest structure{@link https://docs.ccxt.com/en/latest/manual.html#interest-history-structure}
+         */
+        const timeframe = this.safeString (params, 'interval', '1h');
+        if (timeframe === '1m') {
+            throw new BadRequest (this.id + ' fetchOpenInterest() cannot use the 1m timeframe');
+        }
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        if (!market['contract']) {
+            throw new BadRequest (this.id + ' fetchOpenInterest() supports contract markets only');
+        }
+        const defaultSubType = this.safeString (this.options, 'defaultSubType', 'linear');
+        const category = this.safeString (params, 'category', defaultSubType);
```

Comment:
@Dan-krm do we need to re-work our symbols? otherwise why wouldn't we read the subType from the symbol itself?

Reply:
I've added this line `const linear = market['linear'] ? 'linear' : 'inverse';` let me know if that's not what you had in mind

Reply:
@Dan-krm that's exactly what I had in mind, but apparently, it's not working for inverse markets: 
![image](https://user-images.githubusercontent.com/43336371/196141401-be52d3e4-fd01-4062-acab-05c6fd43bdff.png)

Reply:
@carlosmiei I'm receiving this error for both linear and inverse now:
```
[InvalidNonce] bybit {"ret_code":10002,"ret_msg":"invalid request,please check your server timestamp or recv_window param","ext_code":"","result":null,"ext_info":null,"time_now":1666068788802}

    at throwExactlyMatchedException  js/base/Exchange.js:2140              throw new exact[string] (message);                                                 
    at handleErrors                  js/bybit.js:5447                      this.throwExactlyMatchedException (this.exceptions['exact'], errorCode, feedbac…   
    at                               js/base/Exchange.js:598               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…   
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                     
    at async timeout                 js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…   
    at fetch2                        js/base/Exchange.js:1801              return await this.fetch (request['url'], request['method'], request['headers'],…   
    at request                       js/base/Exchange.js:1805              return await this.fetch2 (path, api, method, params, headers, body, config, con…   
    at fetchCurrencies               js/bybit.js:653                       const response = await this.privateGetAssetV1PrivateCoinInfoQuery (params);        
    at loadMarketsHelper             js/base/Exchange.js:624               currencies = await this.fetchCurrencies ()                                         
    at async run                     examples/js/cli.js:231                await exchange.loadMarkets ()                                                      

bybit {"ret_code":10002,"ret_msg":"invalid request,please check your server timestamp or recv_window param","ext_code":"","result":null,"ext_info":null,"time_now":1666068788802}
```

Reply:
@Dan-krm but that's a different error, usually means a difference between your clock and the exchange.

Reply:
@carlosmiei with the current ordering defaultSubType in the exchange options need to be set to inverse for it to work properly, I think removing defaultSubType from fetchOpenInterest is the best approach